### PR TITLE
Fix handling of abstract/virtual in structs

### LINKED ISSUE #6590: C# compiler erroneously allows declarations of abstract members in a struct
The following code compiles, but should  not:

    struct Foo
    {
        public abstract void Bar();
    }

    new Foo().Bar() //TypeLoadException at runtime: 'Abstract method in non-abstract class.'

### PR DESCRIPTION
**Pull request by leppie**
Fixes #6590 and #7026

### CODE REVIEW

**Comment by davkean** [comment]
Thanks for another contribution.

tag @dotnet/roslyn-compiler

**Comment by davkean** [comment]
@dotnet/roslyn-infrastructure  Looks like we've run out of disk space on  dci-ub-fbld-3. @TyOverby can you take a look tomorrow?

**Comment by leppie** [comment]
@dotnet-bot retest prtest/lin/dbg/unit32 please

**Comment by leppie** [comment]
Keep this open for now, found issues for properties as well.

**Comment by leppie** [comment]
And events are joining the party too! Anything else I am missing?

**Comment by dpoeschl** [comment]
Updating @davkean's thread above: I've taken fbld-3 offline, but I haven't investigated what's wrong with it. Subsequent builds are routing to other machines and passing. /cc: @TyOverby

**Comment by davkean** [comment]
@dotnet test eta please

**Comment by davkean** [comment]
Thanks @dpoeschl forgot to take it offline. :(

**Comment by davkean** [comment]
@leppie Just to set expectations, it's Thanksgiving tomorrow in the US - so lots of us are out, it may take until start of next week to get reviewed.

**Comment by leppie** [comment]
@davkean No problem, I am in no rush :D

File: src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs [hunk 1]
```diff
@@ -1672,6 +1672,86 @@ class C
...
         [Fact]
+        public void CS0106ERR_BadMemberFlag05()
+        {
+            var text = @"
+struct Foo
+{
+    public abstract void Bar();
+}
+";
+            DiagnosticsUtils.VerifyErrorsAndGetCompilationWithMscorlib(text,
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs]: Consider following style used by CS0106ERR_BadMemberFlag03 for the baseline. It is easier to read. If you run test with empty argument list for VerifyDiagnostics, the failure output for the unit-test will produce C# code for the argument list that you can simply copy/paste as the baseline.

**Comment by AlekseyTs** [comment]
Consider adding tests for indexers as well.

File: src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs (hunk 1 above)
Comment (leppie) [inline src/Compilers/CSharp/Test/Symbol/Symbols/SymbolErrorTests.cs]: @AlekseyTs: Will do :D

**Comment by leppie** [comment]
retest prtest/win/dbg/unit32 please

**Comment by davkean** [comment]
@AlekseyTs @gafter What's this waiting on?

**Comment by AlekseyTs** [comment]
LGTM

**Comment by davkean** [comment]
retest eta please

**Comment by jaredpar** [comment]
:+1:

**Pull request merged**